### PR TITLE
Implement a Service Server API loosely following the ROS2 API.

### PR DESCRIPTION
**Pull request by aentinger**

### CODE REVIEW

File: src/Node.hpp [hunk 1]
```diff
@@ -87,51 +85,39 @@ class Node
                                                    CanardMicrosecond const rx_timeout_usec,
                                                    OnReceiveCb&& on_receive_cb);
 
+  template <typename T_REQ, typename T_RSP>
+  Service<T_REQ, T_RSP> create_service(CanardPortID const port_id,
+                                       CanardMicrosecond const tx_timeout_usec,
+                                       std::function<T_RSP(T_REQ const &)> on_request_cb);
```
Comment (pavel-kirienko) [inline src/Node.hpp:91]: I think it makes sense, if only for the sake of consistency, to erase the type of the callback the same way you did it for subscribers.

**Review by pavel-kirienko — CHANGES_REQUESTED** [review]

File: src/Node.hpp (hunk 1 above)
Comment (aentinger) [reply src/Node.hpp]: I did in 532ba205dd1b97a667c2307aeab4d8bb7fb1c896, but here's thing that has puzzled me: What type do I use for the callback template parameter when I can not use `auto`, i.e. [here](https://github.com/107-systems/107-Arduino-Cyphal/blob/service-server/src/nodeinfo/NodeInfo.h#L43)? The actual type is this [lambda](https://github.com/107-systems/107-Arduino-Cyphal/blob/service-server/src/nodeinfo/NodeInfo.cpp#L48-L53) but I can hardly declare a auto member variable.

The same issue [here](https://github.com/107-systems/107-Arduino-Cyphal/blob/service-server/src/register/RegisterList.h#L44) and [here](https://github.com/107-systems/107-Arduino-Cyphal/blob/service-server/src/register/RegisterList.h#L49) with lambdas [here](https://github.com/107-systems/107-Arduino-Cyphal/blob/service-server/src/register/RegisterList.cpp#L234-L236) and [here](https://github.com/107-systems/107-Arduino-Cyphal/blob/service-server/src/register/RegisterList.cpp#L242-L245) being the template parameter.

Compile error in all cases is 
```bash
107-Arduino-Cyphal/src/register/RegisterList.h:49:42: error: wrong number of template arguments (2, should be 3)
   Service<TAccessRequest, TAccessResponse> _reg_access_srv;
```

Comment (pavel-kirienko) [reply src/Node.hpp]: That's what you need type erasure for! As far as I see, you should be able to replace your type-aware `shared_ptr<Service<>>` with type-erased `shared_ptr<ServiceBase>` pretty much everywhere, as you did with subscribers. The `create_service()` factory may still return the concrete type, as it is implicitly convertible to `shared_ptr<ServiceBase>`.

Comment (aentinger) [reply src/Node.hpp]: This type erasure stuff is mind-bending. Should be fixed in 6386de0652a1a13f3a6669fafcbafac7879ef532 :heavy_check_mark:

**Pull request merged**